common/src/main/java/com/microsoft/identity/common/internal/commands/parameters/SilentTokenCommandParameters.java

```diff
@@ -72,12 +82,15 @@ private boolean authorityMatchesAccountEnvironment() {
             }
             final AzureActiveDirectoryCloud cloud = AzureActiveDirectory.getAzureActiveDirectoryCloudFromHostName(getAccount().getEnvironment());
             return cloud != null && cloud.getPreferredNetworkHostName().equals(getAuthority().getAuthorityURL().getAuthority());
-        } catch (IOException e) {
+        } catch (final IOException e) {
             Logger.error(
                     TAG + methodName,
                     "Unable to perform cloud discovery",
                     e);
-            return false;
+            throw new TerminalException(
```

Comment:
Asked this question and got an answer from @AdamBJohnsonx offline - this new exception is to avoid signature changes on `validate()` which would have become more invasive